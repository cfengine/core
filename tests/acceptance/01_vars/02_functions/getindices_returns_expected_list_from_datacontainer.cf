#######################################################
#
# Test that getindices() returns the expected list from
# a data container
#
#######################################################

body common control
{
      inputs => { "../../default.cf.sub" };
      bundlesequence  => { default("$(this.promise_filename)") };
      version => "1.0";
}

#######################################################

bundle agent init
{
  vars:
    "data" data => parsejson('{
  "foo": [
          "alpha",
          "bravo"
         ],
  "bar": { "one": "1"  "two": "2" }
}');
}

#######################################################

bundle agent test
{
  meta:
      "test_soft_fail" string => "!any",
        meta => { "redmine7116" };

  vars:
      # expected: one, two
      "values_data" slist => getindices("init.data[bar]");
}

#######################################################

bundle agent check
{
  methods:
      "check"  usebundle => dcs_check_state(test,
                                           "$(this.promise_filename).expected.json",
                                           $(this.promise_filename));
}
