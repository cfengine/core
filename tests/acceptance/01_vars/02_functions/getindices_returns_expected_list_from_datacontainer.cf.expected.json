{
  "values_data": [
    "one",
    "two"
  ]
}
